calc/async_calc_graph.go

```diff
@@ -141,14 +142,18 @@ func (acg *AsyncCalcGraph) loop() {
 			case []api.Update:
 				// Update; send it to the dispatcher.
 				log.Debug("Pulled []KVPair off channel")
-				updStartTime := time.Now()
-				acg.AllUpdDispatcher.OnUpdates(update)
-				summaryUpdateTime.Observe(time.Since(updStartTime).Seconds())
-				// Record stats for the number of messages processed.
-				for _, upd := range update {
+				for i, upd := range update {
```

Comment:
@neiljerram I realised I had put the call to reportHealth() in the wrong place.  Needed to split up the calls to OnUpdates instead.